### PR TITLE
WIP: Added xtodo crate

### PR DESCRIPTION
**Pull request by ZapAnton**
This PR adds a `xtodo` utility, the main function of which is to list unimplemented or outdated exercises on the track it is invoked, by comparing the exercises and their versions on the track with the exercises on the `problem-specifications` repo.

The main motivation is to help new contributors to find an exercise that could be added or updated, so that they have a better chance to submit a PR to the track repo and also to help track maintainers to see the current status of the track.

Originally the utility was planned to be cross-track, but, as stated in the exercism/exercism#4466, not every track keeps the version of the exercises, so the listing of the outdated exercises is limited to the tracks that have some sort of versioning (currently only Rust and Haskell tracks).

Opened as `In Progress`, because not every track is supported, and documentation/error handling could be improved.

If the maintainers deem this utility worthy of being present on the track, then it would be further imporoved, otherwise this PR can be closed.

### CODE REVIEW

**Comment by ZapAnton** [comment]
An example of `xtodo outdated` command result on the Rust track:
```text
Outdated exercises for the Rust track:
 1) leap                      local version: 1.0.0  canonical version: 1.4.0 
 2) reverse-string            local version: 1.0.0  canonical version: 1.1.0 
 3) nth-prime                 local version: 1.0.0  canonical version: 2.1.0 
 4) bob                       local version: 1.2.0  canonical version: 1.4.0 
 5) beer-song                 local version: 0.0.0  canonical version: 2.1.0 
 6) sum-of-multiples          local version: 1.0.0  canonical version: 1.5.0 
 7) grains                    local version: 1.1.0  canonical version: 1.2.0 
 8) series                    local version: 0.1.0  canonical version: 1.0.0 
 9) collatz-conjecture        local version: 1.2.0  canonical version: 1.2.1 
10) diffie-hellman            local version: 0.1.0  canonical version: 1.0.0 
11) isogram                   local version: 1.3.0  canonical version: 1.7.0 
12) say                       local version: 1.1.0  canonical version: 1.2.0 
13) isbn-verifier             local version: 1.1.0  canonical version: 2.7.0 
14) hamming                   local version: 0.0.0  canonical version: 2.1.1 
15) pascals-triangle          local version: 1.3.0  canonical version: 1.5.0 
16) pangram                   local version: 0.0.0  canonical version: 1.4.1 
17) nucleotide-count          local version: 0.0.0  canonical version: 1.3.0 
18) luhn                      local version: 0.0.0  canonical version: 1.3.0 
19) largest-series-product    local version: 1.1.0  canonical version: 1.2.0 
20) word-count                local version: 0.0.0  canonical version: 1.2.0 
21) atbash-cipher             local version: 1.1.0  canonical version: 1.2.0 
22) crypto-square             local version: 0.1.0  canonical version: 3.2.0 
23) rotational-cipher         local version: 1.0.0  canonical version: 1.2.0 
24) simple-cipher             local version: 0.0.0  canonical version: 1.2.0 
25) acronym                   local version: 1.0.0  canonical version: 1.6.0 
26) rna-transcription         local version: 1.0.0  canonical version: 1.3.0 
27) triangle                  local version: 0.0.0  canonical version: 1.2.0 
28) roman-numerals            local version: 1.0.0  canonical version: 1.2.0 
29) all-your-base             local version: 1.0.0  canonical version: 2.3.0 
30) grade-school              local version: 0.0.0  canonical version: 1.0.0 
31) binary-search             local version: 1.1.0  canonical version: 1.2.0 
32) robot-simulator           local version: 2.2.0  canonical version: 3.1.0 
33) queen-attack              local version: 2.1.0  canonical version: 2.2.0 
34) bowling                   local version: 1.0.0  canonical version: 1.2.0 
35) sublist                   local version: 0.0.0  canonical version: 1.1.0 
36) allergies                 local version: 1.1.0  canonical version: 1.2.0 
37) variable-length-quantity  local version: 1.1.0  canonical version: 1.2.0 
38) phone-number              local version: 1.2.0  canonical version: 1.6.1 
39) custom-set                local version: 1.0.1  canonical version: 1.3.0 
40) alphametics               local version: 1.0.0  canonical version: 1.3.0 
41) two-bucket                local version: 1.0.0  canonical version: 1.4.0 
42) pig-latin                 local version: 1.0.0  canonical version: 1.2.0 
43) grep                      local version: 1.1.0  canonical version: 1.2.0 
44) scale-generator           local version: 1.0.0  canonical version: 2.0.0 
45) anagram                   local version: 0.0.0  canonical version: 1.4.0 
46) protein-translation       local version: 0.1.0  canonical version: 1.1.1 
47) book-store                local version: 1.3.0  canonical version: 1.4.0 
48) ocr-numbers               local version: 0.0.0  canonical version: 1.2.0 
49) forth                     local version: 0.0.0  canonical version: 1.7.0
```

**Comment by ZapAnton** [comment]
An example of `xtodo missing` command result on the Rust track:

```text
Unimplemented exercises for the Rust track:
1) flatten-array
2) hangman
3) change
4) kindergarten-garden
5) yacht
6) counter
7) ledger
8) error-handling
9) binary-search-tree
10) bank-account
11) meetup
12) food-chain
13) twelve-days
14) pov
15) rest-api
16) matrix
17) go-counting
18) rational-numbers
19) complex-numbers
20) markdown
21) trinary
22) darts
23) linked-list
24) connect
25) tree-building
26) secret-handshake
27) word-search
28) strain
29) list-ops
30) house
31) zebra-puzzle
32) lens-person
33) binary
34) high-scores
35) zipper
36) sgf-parsing
37) affine-cipher
38) point-mutations
39) transpose
40) octal
```

**Comment by petertseng** [comment]
The tl;dr is that I personally find it unlikely, but I will not object to other maintainers' acceptances.

First, a general thought I have:

If the track maintainers for a given track (here, Rust track) take on a piece of code, it is a sort of adoption, and the maintainers will be choosing to either be involved in the maintenance of that code thereafter... or not to be involved, but then they accept that the code may eventually bit-rot.

How does that apply to this code?

I'll start with matters that are actually relevant to all Rust track maintainers: It does not make much sense to be responsible for code that deals with any other track (in this current case, the Haskell track, but true of any other track). "The incentives are not aligned" for the Rust track maintainers to do that (I think there is very little, if at all).

So, fundamentally, I think **if this tool is meant to be usable on many tracks, then it cannot make sense for any single track's maintainers to be the only maintainers for it**.

So that's one choice. It's likely the tool is meant for multiple tracks, in which case it shouldn't be put in one track.

The other choice is to make it only applicable to the Rust track. What would happen then?

For this part, I think I have no reason applicable to all Rust track maintainers, so I can only speak of matters that concern me only: It is unlikely I will ever personally run this code, since I would instead run https://github.com/petertseng/exercism-problem-specifications/tree/up-to-date/up-to-date than xtodo. The reason is I will be able to make changes to up-to-date far faster than I could to xtodo. That decreases the value that xtodo presents to me.

That means I still would not have much incentive to take on the xtodo code, so I would have to leave the review of that to others.

**Comment by petertseng** [comment]
I think you already made the outdated list exclude exercises that are deprecated on the track, which is good.

You should also exclude from the unimplemented list any exercise that is deprecated globally for exercism.

For example, octal is https://github.com/exercism/problem-specifications/blob/master/exercises/octal/.deprecated .
